### PR TITLE
add minion movement action

### PR DESCRIPTION
**Pull request by electrovir**
create user on initial game state

### CODE REVIEW

File: types/game-action.d.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ export type AddPlayerAction = {
     readonly player: Readonly<Player>,
 };
 
-export type CreateUserAction = {
-    readonly type: 'CREATE_USER',
+export type MoveMinionAction = {
+    readonly type: 'MOVE_MINION',
+    readonly minionId: string,
+    readonly movementDirection: 'up' | 'down' | 'left' | 'right';
```
Comment (lastmjs) [inline types/game-action.d.ts:19]: These should be constants

File: types/minion.d.ts [hunk 2]
```diff
@@ -2,4 +2,7 @@ export type Minion = {
     readonly x: number,
     readonly y: number,
     readonly playerId: string,
-};
\ No newline at end of file
+    readonly minionId: string,
+};
+
+export type Minions = {[minionId: string]: Minion};
```
Comment (lastmjs) [inline types/minion.d.ts:8]: Make everything readonly

File: frontend/modules/game-store.ts [hunk 3]
```diff
@@ -45,20 +52,38 @@ function gameStoreReducer(state: Readonly<GameState>, action: Readonly<GameActio
                     [action.player.id]: action.player
                 },
             };
-        case 'CREATE_USER':
-            const user = generateRandomUserPlayer();
+
+        case 'MOVE_MINION':
+            const minion = state.minions[action.minionId];
+            const nextCoords = {
```
Comment (lastmjs) [inline frontend/modules/game-store.ts:58]: Do this immutably, no switch

File: types/game-action.d.ts (hunk 1 above)
Comment (electrovir) [reply types/game-action.d.ts:19]: Meaning uppercase.

**Pull request merged**